app/src/main/java/com/github/se/icebreakrr/model/tags/TagsRepository.kt

```diff
@@ -0,0 +1,241 @@
+            var tagsCategory = TagsCategory()
+            firestoreToTags(
+                documentSnapshot,
+                { tagCategoryCallback -> tagsCategory = tagCategoryCallback },
+                { e: Exception ->
+                  Log.e(
+                      "TagsRepository",
+                      "[addCategory] error while converting firebase to Tags : $e")
+                  onFailure(e)
+                })
+            val subtags = tagsCategory.subtags.toMutableList()
+            subtags.addAll(subcategories.filter { it !in subtags })
+            db.collection(collectionPath)
+                .document(categoryName)
+                .update("subtags", subtags.toList())
+                .addOnSuccessListener {}
+                .addOnFailureListener { exception: Exception ->
+                  Log.e("TagsRepository", "[addCategory] Could not update the document: $exception")
+                  onFailure(exception)
+                }
+          }
+          .addOnFailureListener { e: Exception ->
+            Log.e("TagsRepository", "[addCategory] fail to get the document :  $e")
+            onFailure(e)
+          }
+    } else {
+      db.collection(collectionPath)
+          .document(categoryName)
+          .set(TagsCategory(categoryName, color, subcategories))
+          .addOnSuccessListener {}
+          .addOnFailureListener { e: Exception ->
+            Log.e("TagsRepository", "[addCategory] could not set the new category :  $e")
+            onFailure(e)
+          }
+    }
+  }
+
+  /**
+   * use this function to delete a tag in a certain category. If the tag isn't in the category, or
+   * the category doesn't exist, the method does nothing.
+   *
+   * @param onFailure : happens if the tags can't be fetch from the database or if the update of the
+   *   database fails
+   * @param name : name of the tag to delete
+   * @param category : category in which the tag you want to suppress is
+   */
+  fun deleteTag(onFailure: (Exception) -> Unit, name: String, category: CategoryString) {
+    getAllTags(
+        { tagList: List<TagsCategory> ->
+          val tagCategory = tagList.find { it.name == category.name }
+          if (tagCategory != null && tagCategory.subtags.contains(name)) {
+            db.collection(collectionPath)
+                .document(category.name)
+                .update("subtags", tagCategory.subtags.filter { it != name })
+                .addOnSuccessListener {}
+                .addOnFailureListener { e: Exception ->
+                  Log.e("TagsRepository", "[deleteTag] Failed to update the document : $e")
+                  onFailure(e)
+                }
+          }
+          Log.d(
+              "TagsRepository",
+              "[deleteTag] the tag wasn't in the database or the category doesn't exist : $name")
+        },
+        { Log.e("TagsRepository", "[deleteTag] error while getting tags : $it") })
+  }
```

Comment:
Nitpick: try expliciting the parameters of getAllTags to make code more readable

```
getAllTags(
   onSuccess=..., 
   onFailure=...
) 
```